### DISCUSSION TITLE
Caching build folder in GitHub Actions does not speed up build

### DISCUSSION THREAD

**Discussion by kjy5** [pyinstaller/pyinstaller#8274] [Help] [answered]
I'm using PyInstlaler in a GitHub action to do automatic build testing, but the process is very slow. I tried to `cache` the build folder to see if successive runs could be sped up, but this didn't seem to be doing anything. Are there any other files/folders I should be caching to help speed up the build process?

Thanks!

**Comment by bwoodsend** [comment]
PyInstaller's caching is done based on files' last modified timestamps. The process of GitHub stashing and restoring the `build` directory will mess with those timestamps, as will cloning/checkout-ing your code. I don't see how successful caching here could ever work.

**Reply by kjy5** [reply]
Hmm ok. Is there anything that can be done to speed up the build process in CI tools? Is there a dry-run option or something that can be done? The point is just to enforce that the package still builds on PRs.

**Reply by bwoodsend** [reply]
Not really. Prune any overgrown/optional dependency ridden packages out of your dependency tree, stick to onedir mode, avoid using flags like `--collect-all` when a mere `--hidden-import` or `--copy-medata` is all you need, break large projects up into lots of little ones if possible.

> Is there a dry-run option or something that can be done? The point is just to enforce that the package still builds on PRs.

What would a dry run mode do that actually verifies anything?

**Reply by rokm** [reply]
> > Is there a dry-run option or something that can be done? The point is just to enforce that the package still builds on PRs.
> 
> What would a dry run mode do that actually verifies anything?

And for that matter, a build-only workflow has very limited indicative value. You could run a build in an environment completely devoid of all dependent packages, and it would still succeed. But the resulting frozen application will fail to run due to missing module(s).

**Reply by kjy5** [reply, marked as answer]
I suppose that's true for Python / PyInstaller. For other platforms and languages I've been doing them as sanity checks since builds can fail due to issues with code.

In the future, I will be doing automated releases which will be built via Actions, but I suppose the speed doesn't really matter on those.

Thanks!
